package.json

```diff
@@ -17,25 +18,37 @@
     "test": "tests"
   },
   "scripts": {
-    "lint": "gulp lint",
-    "test": "gulp test && coffee tests/e2e/test.coffee",
-    "build": "gulp build",
-    "prepublish": "npm test && npm run build",
+    "test:unit": "mocha -r ts-node/register --reporter spec tests/**/*.spec.ts",
+    "test:e2e": "ts-node tests/e2e/test.ts",
+    "lint": "tslint lib/**/*.ts",
+    "test": "npm run lint && npm run build && npm run test:unit && npm run test:e2e",
+    "build": "rimraf ./build && npm run prettify && tsc && npm run readme",
+    "prepublish": "npm test",
+    "prettify": "prettier --use-tabs --single-quote --write lib/*.ts tests/**/*.ts",
     "readme": "jsdoc2md --template doc/README.hbs build/settings.js > README.md"
   },
   "author": "Juan Cruz Viotti <juan@resin.io>",
   "license": "Apache-2.0",
   "devDependencies": {
-    "bluebird": "^3.0.0",
-    "coffee-script": "~1.12.5",
-    "gulp": "^3.8.10",
-    "gulp-coffee": "^2.2.0",
-    "gulp-coffeelint": "^0.6.0",
-    "gulp-mocha": "^3.0.0",
-    "gulp-util": "^3.0.1",
-    "jsdoc-to-markdown": "^1.1.1",
-    "mocha": "^3.0.0",
-    "mochainon": "^1.0.0",
+    "@types/bluebird": "^3.5.8",
+    "@types/chai": "^4.0.2",
+    "@types/chai-as-promised": "0.0.31",
+    "@types/common-tags": "^1.2.5",
+    "@types/js-yaml": "^3.9.0",
+    "@types/lodash": "^4.14.71",
```

Comment:
`@types` for js-yaml and lodash should live in `dependencies`, not `devDependencies`, because I think upstream TS packages need to install those too to be able to use this package.